resources/knative-eventing/charts/event-mesh-dashboard/templates/event-mesh-latency-dashboard.yaml

```diff
@@ -0,0 +1,603 @@
+              "hide": false,
+              "legendFormat": "{{`{{source_workload_namespace}}`}}/{{`{{source_workload}}`}} -> {{`{{destination_workload_namespace}}`}}/{{`{{destination_workload}}`}} (p99)",
+              "refId": "A"
+            }
+          ],
+          "thresholds": [],
+          "timeFrom": null,
+          "timeRegions": [],
+          "timeShift": null,
+          "title": "HTTP Event Source -> Source Channel",
+          "tooltip": {
+            "shared": true,
+            "sort": 0,
+            "value_type": "individual"
+          },
+          "type": "graph",
+          "xaxis": {
+            "buckets": null,
+            "mode": "time",
+            "name": null,
+            "show": true,
+            "values": []
+          },
+          "yaxes": [
+            {
+              "format": "s",
+              "label": null,
+              "logBase": 1,
+              "max": null,
+              "min": null,
+              "show": true
+            },
+            {
+              "format": "short",
+              "label": null,
+              "logBase": 1,
+              "max": null,
+              "min": null,
+              "show": false
+            }
+          ],
+          "yaxis": {
+            "align": false,
+            "alignLevel": null
+          }
+        },
+        {
+          "aliasColors": {},
+          "bars": false,
+          "dashLength": 10,
+          "dashes": false,
+          "datasource": null,
+          "fill": 1,
+          "fillGradient": 0,
+          "gridPos": {
+            "h": 8,
+            "w": 12,
+            "x": 0,
+            "y": 9
+          },
+          "hiddenSeries": false,
+          "id": 6,
+          "legend": {
+            "avg": false,
+            "current": false,
+            "max": false,
+            "min": false,
+            "show": true,
+            "total": false,
+            "values": false
+          },
+          "lines": true,
+          "linewidth": 1,
+          "nullPointMode": "null",
+          "options": {
+            "dataLinks": []
+          },
+          "percentage": false,
+          "pointradius": 2,
+          "points": false,
+          "renderer": "flot",
+          "seriesOverrides": [],
+          "spaceLength": 10,
+          "stack": false,
+          "steppedLine": false,
+          "targets": [
+            {
+              "expr": "histogram_quantile(0.99, sum(rate(istio_request_duration_seconds_bucket{source_workload=~\"[A-Za-z]+-ch-dispatcher\",destination_workload=\"default-broker-ingress\"}[5m])) by (le,source_workload_namespace,source_workload,destination_workload_namespace,destination_workload))",
```

Comment:
FYI:
- Natss Channel Dispatcher has the `app=natss-ch-dispatcher` label which should already be exposed as a prometheus label called `source_app`. Why not use this instead of `source_workload` ?

Here are the labels for `natss-ch-dispatcher`:

```shell
$ k get po -n knative-eventing -l app=natss-ch-dispatcher --show-labels
NAME                                   READY   STATUS    RESTARTS   AGE    LABELS
natss-ch-dispatcher-76fb89fb7f-rdvtj   2/2     Running   0          3d7h   app=natss-ch-dispatcher,kyma-project.io/dashboard=event-mesh,messaging.knative.dev/channel=natss-channel,messaging.knative.dev/role=dispatcher,pod-template-hash=76fb89fb7f,rand=fTUvOI,security.istio.io/tlsMode=istio
```

For `knative-kafka`, the pod names and labels look as following:

```shell
$ k get po -n knative-eventing --show-labels|grep kafka
test-kafka-channel-knative-eventing-channel-785d44f659-pqrqn      0/2     Init:0/1    0          3s      app=test-kafka-channel-knative-eventing-channel,pod-template-hash=785d44f659,security.istio.io/tlsMode=istio
test-kafka-channel-knative-eventing-dispatcher-678854ffb6-bn29c   0/2     Init:0/1    0          3s      app=test-kafka-channel-knative-eventing-dispatcher,pod-template-hash=678854ffb6,security.istio.io/tlsMode=istio
```

If we need a custom label (to write our prometheus expression) which is not exposed by istio, we can expose that as a prometheus label as described [here](https://istio.io/docs/reference/config/policy-and-telemetry/metrics/)

Reply:
We definitely need better metrics. Let's create follow up tasks.